default-plugins/tab-bar/src/main.rs

```diff
@@ -70,8 +84,20 @@ impl ZellijPlugin for State {
             self.mode_info.capabilities,
         );
         let mut s = String::new();
-        for bar_part in tab_line {
-            s = format!("{}{}", s, bar_part.part);
+        let mut len_cnt = 0;
+        for (idx, bar_part) in tab_line.iter().enumerate() {
+            s = format!("{}{}", s, &bar_part.part);
+
+            if self.should_render
+                && self.mouse_click_pos > len_cnt
+                && self.mouse_click_pos <= len_cnt + bar_part.len
+                && idx > 1
+            {
+                // First two elements of tab_line are "Zellij" and session name, hence the idx > 1 condition.
+                // Tabs are indexed starting from 1, therefore we need subtract 1 below.
+                switch_tab_to(TryInto::<u32>::try_into(idx).unwrap() - 1);
+            }
+            len_cnt += bar_part.len;
```

Comment:
As the width of the window shrinks, parts of the tab bar get either merged or left off. This logic can break down a little there, but so do some other (visual) things and these are bugs we've been living with. 

Overall then, this is fine as is, but if you want to also add scrolling with the mouse-wheel to change tabs, that might provide a nice, user-friendly way to change tabs with the mouse even when they don't all fit on the screen.